agent/testflinger_agent/agent.py

```diff
@@ -278,11 +306,15 @@ def process_jobs(self):
                             exit_event = TestEvent.RECOVERY_FAIL
                         else:
                             exit_event = TestEvent(phase + "_fail")
-                        event_emitter.emit_event(exit_event)
+                        detail = ""
                         if phase == "provision":
                             self.client.post_provision_log(
                                 job.job_id, exit_code, exit_event
                             )
+                            detail = parse_provision_error_logs(
+                                provision_error_log_path
+                            )
+                        event_emitter.emit_event(exit_event, detail)
```

Comment:
I don't see any reason we should limit this to *only* happen during the provision phase. Yes, we're focusing on the provision phase of maas for now, because that's a big target for us. But there's a chance that we could capture other kinds of unexpected errors at runtime from other phases if we just do this as a generic step for each phase and add it to the detail if we find some exception that gets propagated back to the agent in this way. It's certainly much MORE likely to happen during the provision phase, but anything is possible, and it might help us later if we unrestrict it.